### PR TITLE
Fixed Some issues in Can_SetControllerMode Function

### PR DESCRIPTION
**Pull request by AAdelBayoumi**
-when request is accepted the function should return E_OK
-Code duplication: fix via Base-Address variable
-messing default case
-reporting error for an allowed transitions sleep->sleep
-some code cleaning

### CODE REVIEW

File: Software/bsw/static/Mcal/CAN/src/Can.c [hunk 1]
```diff
@@ -1343,163 +1344,71 @@ Std_ReturnType Can_SetControllerMode( uint8 Controller, Can_ControllerStateType
                 if(DisableCnt[0U]==0U)
```
Comment (ahmedwahdan) [inline Software/bsw/static/Mcal/CAN/src/Can.c:1344]: You should use controller  as index.

Comment (ahmedwahdan) [inline Software/bsw/static/Mcal/CAN/src/Can.c:1344]: For the the disable counter I believe the following:
Stopped request:
Stopping steps should be execute once and only the counter will be only incremented afterwards.
Starting request:
The counter should be decremented first and we check if we reach zero, only then we enable the CAN.

File: Software/bsw/static/Mcal/CAN/src/Can.c [hunk 2]
```diff
@@ -1343,163 +1344,71 @@ Std_ReturnType Can_SetControllerMode( uint8 Controller, Can_ControllerStateType
...
                     /* Enables individual CAN controller interrupt sources */
-                    HWREG(CAN0_BASE + CAN_O_CTL) |= CAN_INT_MASTER | CAN_INT_ERROR | CAN_INT_STATUS;
+                    HWREG(ui32BaseAddress + CAN_O_CTL) |= CAN_INT_MASTER | CAN_INT_ERROR | CAN_INT_STATUS;
                     /* Enable the general interrupt.*/
-                    HWREG(g_pui32EnRegs[(INT_CAN0 - 16) / 32]) =(1 << ((INT_CAN0 - 16) & 31));
+                    CANIntEnable(ui32BaseAddress,CAN_CTL_EIE | CAN_CTL_SIE | CAN_CTL_IE);
+                    //HWREG(g_pui32EnRegs[(INT_CAN0 - 16) / 32]) =(1 << ((INT_CAN0 - 16) & 31));
                     /* Enables the CAN controller for message processing.  Once enabled, the
                   controller automatically transmits any pending frames, and processes any
                   received frames.*/
```
Comment (SaharElnagar) [inline Software/bsw/static/Mcal/CAN/src/Can.c:1353]: Why do we enable interrupt in Can_SetControllerMode and not in Can_EnableControllerInterrupts  ?

Comment (ahmedwahdan) [reply Software/bsw/static/Mcal/CAN/src/Can.c:1353]: Please have a look on the latest changes on DevCAN, I have updated the implementation

**Pull request merged**